test-data/unit/check-unreachable-code.test

```diff
@@ -915,15 +915,15 @@ def expect_str(x: str) -> str: pass
 
 x: int
 if False:
-    assert False
+    assert False  # E: Statement is unreachable
     reveal_type(x)
 
 if False:
-    raise Exception()
+    raise Exception()  # E: Statement is unreachable
     reveal_type(x)
 
 if False:
-    assert_never(x)
+    assert_never(x)  # E: Statement is unreachable
```

Comment:
Here `assert_never` is used to generate an error only if the statement is reachable, so this seems like a false positive. If we special case `reveal_type` as an empty statement this would continue to work as before.

Reply:
Oh I see, I was confused to. But, it is `reveal_type(x)` that is unreachable. Right now it generates an error on a wrong line.

Reply:
I actually thought about checking `isinstance(s.expr.analyzed, RevealExpr)`, but then I though that this is out of scope. 🙂